crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -771,19 +624,7 @@ impl Stage for SystemStage {
             self.world_id = Some(world.id());
         }
 
-        if self.systems_modified {
-            self.initialize_systems(world);
-            self.rebuild_orders_and_dependencies();
-            self.systems_modified = false;
-            self.executor.rebuild_cached_data(&self.parallel);
-            self.executor_modified = false;
-            if world.contains_resource::<ReportExecutionOrderAmbiguities>() {
-                self.report_ambiguities(world);
-            }
-        } else if self.executor_modified {
-            self.executor.rebuild_cached_data(&self.parallel);
-            self.executor_modified = false;
-        }
+        self.initialize(world);
```

Comment:
This functionality is useful to the public, so I refactored it out into its own method.

In the long-term I'd prefer a type-safe pattern for this sort of initialized vs uninitailized state, but that's way out of scope.